<launch>        
    <!--Args-->
    <arg name="bag_freq" default="5"/>
    <arg name="bag_path" default="$(find platform_detector)/bags"/>
    <arg name="old_bags" default="$(arg bag_path)/old"/>
    <arg name="14_07_bags" default="$(arg bag_path)/14_07_2022"/>
    <arg name="15_03_bags" default="$(arg bag_path)/15_03_2022-FADEUP"/>
    <arg name="20_07_bags" default="$(arg bag_path)/20-21-22_07_2022"/>
    <arg name="all_bags" default="*.bag"/>
    <arg name="namespace" value=""/>

    <!-- Params -->
    <param name="namespace" value="$(arg namespace)"/>
    <param name="use_sim_time" value="true" />
    <param name="sync_sensors" value="true"/>
    <param name="dataset_path" value="$(find platform_detector)/datasets/dummy/"/>
    <param name="dataset_img_format" value=".bmp"/>
    
    <!-- Variable args-->
    <!-- if namespace is needed put the namespace in topic names-->
    <arg name="topics" default="
        $(arg namespace)/usb_cam/image_raw/camera_info 
        $(arg namespace)/usb_cam/image_raw/compressed 
        $(arg namespace)/image_raw/compressed 
        $(arg namespace)/image_raw/camera_info 
        $(arg namespace)/os_cloud_node/points"
    />
    <arg name="renamed_topics" default="
        $(arg namespace)/usb_cam/image_raw/compressed:=$(arg namespace)/old/usb_cam/image_raw/compressed
        $(arg namespace)/image_raw/compressed:=$(arg namespace)/old/image_raw/compressed
        $(arg namespace)/os_cloud_node/points:=$(arg namespace)/old/os_cloud_node/points
    "/>
    <arg name="time_filtered_topics" default="
        $(arg namespace)/usb_cam/image_raw/compressed 
        $(arg namespace)/image_raw/compressed
        $(arg namespace)/os_cloud_node/points
    "/>

    <!-- Params -->
    <param name="pcl_data_topic" value="$(arg namespace)/os_cloud_node/points"/>
    <param name="thermal_data_topic" value="$(arg namespace)/usb_cam/image_raw/compressed"/>
    <param name="thermal_info_topic" value="$(arg namespace)/usb_cam/image_raw/camera_info"/>
    <param name="rgb_data_topic" value="$(arg namespace)/image_raw/compressed"/>
    <param name="rgb_info_topic" value="$(arg namespace)/image_raw/camera_info"/>

    <!-- Launch bags-->
    <node pkg="rosbag" type="play" name="rosbag" output="screen" args="
    $(arg old_bags)/test_16.bag
    $(arg old_bags)/test_17.bag
    $(arg renamed_topics)
    --topics $(arg topics)
    --rate $(arg bag_freq)
    --clock
    "/>

    <!-- Launch time sync node to synchronize sensors timestamps -->
    <include file="$(find platform_detector)/launch/time_sync.launch" />
    
    <!-- Important transforms-->
    <!--<node pkg="rosbag" type="play" name="bags" output="screen" args=" clock $(arg bag_path)$(arg test_18) $(arg bag_path)$(arg test_19) $(arg bag_path)$(arg test_20) $(arg bag_path)$(arg test_21) loop"/>-->
    <node pkg="tf" type="static_transform_publisher" name="cam2map" args="0 0 0 0 0 0 1  map cam 30" />
    <node pkg="tf" type="static_transform_publisher" name="camera_frame2cam" args="0 0 0 0 0 0 1  cam camera_frame 30" />
    <!-- RIGHT -->
    <node pkg="tf" type="static_transform_publisher" name="lidar2cam" args="0.119842 -0.22668 0.0570006 0.5 0.5 0.5 0.5  camera_frame os_sensor 100" />
    <!--<node pkg="tf" type="static_transform_publisher" name="lidar2cam" args="-0.119842 0.22668 -0.0570006 0.5 0.5 0.5 0.5  camera_frame os_sensor 100" />-->
    <!--<node pkg="tf" type="static_transform_publisher" name="lidar2cam" args="0.0570006 0.119842 -0.22668 0.5 0.5 0.5 0.5  camera_frame os_sensor 100" />-->
    <node pkg="tf" type="static_transform_publisher" name="lidar2os_sensor" args="0 0 0 0 0 0 1  os_sensor lidar 30" />

    <!-- Launch fuser node-->
    <node pkg="platform_detector" type="fuser" name="fuser" output="screen"/>

    <!-- Launch Rviz -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find platform_detector)/rviz/visualize.rviz" />
</launch>
